src/screenshot_gui.cpp

```diff
@@ -72,3 +72,21 @@ void ShowScreenshotWindow()
 	CloseWindowById(WC_SCREENSHOT, 0);
 	new ScreenshotWindow(&_screenshot_window_desc);
 }
+
+/**
+ * Hides or shows the screenshot window when taking a screenshot.
+ * @param hidden Are we hiding the window or showing it again after the screenshot is taken?
+ */
+void SetScreenshotWindowHidden(bool hidden)
+{
+	ScreenshotWindow *scw = (ScreenshotWindow *)FindWindowById(WC_SCREENSHOT, 0);
+	if (scw != nullptr) {
+		if (hidden) {
+			scw->SetDirty();
```

Comment:
The content of the if/else is a bit weird. In the first body you do `SetDirty` -> `Change left`, in the second you do `Change left` -> `SetDirty`. From a reader perspective, this suggests the order matter or something is going on. Where in reality the order really doesn't matter.

I would suggest to move the `SetDirty` outside of the `if`. Normally you do it after changing `left` , as that just reads better: change the position, mark it as dirty. The reverse: mark it as dirty, change position, feels .. not natural. But code-wise the order doesn't matter.

Reply:
The order does matter, it's not accidental.

Reply:
huh? That .. needs a comment in the code for sure ;)

Reply:
In trunk, Window::SetDirty marks the area of the screen which the window is over as dirty. If the window is off-screen, then Window::SetDirty will have no effect.